CenterForOpenScience/osf.io
# Adding validation to APIv2 unregistered contributor email [ENG-56]

## Purpose

When submitting an unregistered contributor, the email is not validated against the list of restricted domains.

## Changes


- api/nodes/serializers.py NodeContributorsCreateSerializer.validate_data() added email validation to the data validation method. 
- api_tests/nodes/views/test_node_contributors_list.py TestNodeContributorAdd.test_unregistered_contributor_invalid_email. Added test for invalid contributor emails. 

## QA Notes


Test creating an unregistered contributor POST v2/nodes/guid/contributors with emails that are disallowed. Verify that you get an error instructing that your email domain is blacklisted. See list of unallowed domains in /website/settings/defaults.py BLACKLISTED_DOMAINS.

## Documentation



## Side Effects



## Ticket

https://openscience.atlassian.net/browse/ENG-56
